typings/index.d.ts

```diff
@@ -4289,8 +4289,8 @@ export type MessageReactionResolvable =
 
 export interface MessageReference {
   channelId: Snowflake;
-  guildId: Snowflake;
-  messageId: Snowflake | null;
+  guildId: Snowflake | undefined;
+  messageId: Snowflake | undefined;
```

Comment:
```suggestion
  guildId?: Snowflake;
  messageId?: Snowflake;
```

Reply:
I'd rather go with the latter because: https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#exact-optional-property-types

Reply:
I believe these should stay as undefined because all 3 properties will always be present, but be undefined if we don’t get that data from Discord. It’s something worth looking into for v14 but for now I think this would be more accurate.

Reply:
there may be other places this change needs applying then, i seem to remember a discussion a while ago about just prefering `?: ...` over `| undefined`

Reply:
The two types mean different things though. `?:` is for missing parameters and undefined is for, well, undefined parameters. If you want I can go through the file to ensure these are correctly set, I believe that would still be within the scope of the PR

Reply:
yeah i know they are different, i just seem to remember a discussion a while back about preferring `?:` for whatever reason

Reply:
Are you sure you're not confusing the jsdocs and the types?

Reply:
[here] this is the discussion i'm talking about

if you view at [this commit](https://github.com/discordjs/discord.js/blob/967b533e9d5f08e2be50698fb98d38820d29d88b/src/structures/MessageEmbed.js) said properties are always present, but either a string or undefined

Reply:
> [here](https://github.com/discordjs/discord.js/pull/3822#pullrequestreview-363274200) this is the discussion i'm talking about

What @SpaceEEC is no longer valid as stated in the link iCrawl sent

Reply:
Yes, but we will enable this option, so we def should go with what it is now or even do both. Preferably both.

```suggestion
  guildId?: Snowflake | undefined;
  messageId?: Snowflake | undefined;
```

Reply:
> Yes, but we will enable this option, so we def should go with what it is now or even do both. Preferably both.

Why both though? They will never be actually missing

Reply:
Ah, I read that wrong, we good then 👍